rust/opendp/src/traits.rs

```diff
@@ -41,15 +41,35 @@ impl_is_not_continuous!(u8, u16, u32, u64, u128, i8, i16, i32, i64, i128, isize,
 pub trait DistanceConstant: 'static + Clone + DistanceCast + Div<Output=Self> + Mul<Output=Self> + PartialOrd {}
 impl<T: 'static + Clone + DistanceCast + Div<Output=Self> + Mul<Output=Self> + PartialOrd> DistanceConstant for T {}
 
+// TODO: Maybe this should be renamed to something more specific to budgeting, and add negative checks? -Mike
```

Comment:
The motivation for making it more specific is because at some point I think we want it to fail if it goes negative, which is a behavior that is specific to budgets. If we don't capture that behavior, then PartialSub is fine if we don't care about returning the details of why the sub failed.